microservices/provision/src/Services/Provision.php

```diff
@@ -2,36 +2,138 @@
 
 namespace Services;
 
+use Ivoz\Core\Application\Service\EntityTools;
+use Ivoz\Provider\Domain\Model\Brand\BrandDto;
+use Ivoz\Provider\Domain\Model\Brand\BrandInterface;
+use Ivoz\Provider\Domain\Model\Brand\BrandRepository;
+use Ivoz\Provider\Domain\Model\Company\CompanyDto;
+use Ivoz\Provider\Domain\Model\Company\CompanyInterface;
+use Ivoz\Provider\Domain\Model\Company\CompanyRepository;
+use Ivoz\Provider\Domain\Model\Extension\ExtensionDto;
+use Ivoz\Provider\Domain\Model\Language\LanguageDto;
+use Ivoz\Provider\Domain\Model\Language\LanguageInterface;
+use Ivoz\Provider\Domain\Model\Terminal\TerminalDto;
+use Ivoz\Provider\Domain\Model\Terminal\TerminalInterface;
+use Ivoz\Provider\Domain\Model\Terminal\TerminalRepository;
+use Ivoz\Provider\Domain\Model\TerminalModel\TerminalModelDto;
 use Ivoz\Provider\Domain\Model\TerminalModel\TerminalModelInterface;
 use Ivoz\Provider\Domain\Model\TerminalModel\TerminalModelRepository;
+use Ivoz\Provider\Domain\Model\User\UserDto;
+use Ivoz\Provider\Domain\Model\User\UserInterface;
+use Ivoz\Provider\Domain\Model\Extension\ExtensionInterface;
+use Ivoz\Provider\Domain\Model\User\UserRepository;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 
 /** @psalm-suppress PropertyNotSetInConstructor */
 class Provision extends AbstractController
 {
     private string $storagePath;
 
+
     public function __construct(
+        private EntityTools $entityTools,
         private TerminalModelRepository $terminalModelRepository,
+        private TerminalRepository $terminalRepository,
+        private CompanyRepository $companyRepository,
+        private UserRepository $userRepository,
+        private BrandRepository $brandRepository,
+        private ?TerminalModelDto $terminalModelDto,
+        private ?TerminalDto $terminalDto,
+        private ?UserDto $userDto,
+        private ?BrandDto $brandDto,
+        private ?ExtensionDto $extensionDto,
+        private ?LanguageDto $languageDto,
+        private ?CompanyDto $companyDto
     ) {
         $this->storagePath = (string) $_ENV['STORAGE_PATH'];
     }
 
-    public function execute(string $genericUrlPattern): bool|string
-    {
-        $terminalModel = $this
-            ->terminalModelRepository
-            ->findOneByGenericUrlPattern($genericUrlPattern);
+    public function execute(
+        string $genericUrlPattern,
+        ?string $specificUrlPattern = null
+    ): bool|string {
+        $isHttps = isset($_SERVER['HTTPS']) && !empty($_SERVER['HTTPS']);
```

Comment:
this shoud go into the controller because it's part of infrastructure (every $_SERVER, $_POST, etc. access)